com.unity.render-pipelines.universal/ShaderLibrary/Shadows.hlsl

```diff
@@ -398,7 +398,11 @@ half MainLightShadow(float4 shadowCoord, float3 positionWS, half4 shadowMask, ha
     half shadowFade = half(1.0);
 #endif
 
+#if defined(LIGHTMAP_SHADOW_MIXING) || defined(SHADOWS_SHADOWMASK) || !defined(_MAIN_LIGHT_SHADOWS_SCREEN)
     return MixRealtimeAndBakedShadows(realtimeShadow, bakedShadow, shadowFade);
```

Comment:
@phi-lira the main goal of this PR is to fix a fugbuz.

And,
While I was working on far-distance shadows using my another screen space shadows renderer feature, the cascade distance faded out those shadows although there are far distance shadows in the scene.
I thought it needs to exclude fade-out in getting shadows from screen space shadow texture when it didn't have baked shadows. This is my opinion and if this idea had problems in the current situation. let me revert it to the original codes. and I'll request another PR for it.

Reply:
@lukaschod can you have a look at this?

Reply:
I am not sure I undestand fully the intention behind the macro usage here. @Seongdae could you eleborate a bit more, as seems like u are removing shadow fade in case of no mixed lighting or am I wron?

Reply:
Move shadow fade to screen space shadows pass when screen space shadows' feature was enabled in case of no mixed lighting.

In this hackweek, I tried to make a custom screen space shadows' feature for shadowing large-scale, and it could cast shadows on anywhere, but it was faded out beyond the distance urp set. At that moment I though any user can also have trouble with it by their own shadowing solution.

Reply:
k I understand I would prefer to avoid adding screen space logic into this function. As we already have user pain where we have defines that change completetly functionality without users aware of it.
Maybe we can simply have ScreenSpaceShadows.shader instead of calling MainLightRealtimeShadow use MainLightRealtimeShadow in case of no mixed lighting?

Reply:
@lukaschod let me think about it and request another PR. the code will be restored in this PR.